app/client/src/widgets/DropdownWidget.tsx

```diff
@@ -101,7 +101,9 @@ class DropdownWidget extends BaseWidget<DropdownWidgetProps, WidgetState> {
   onOptionSelected = (selectedOption: DropdownOption) => {
     let isChanged = true;
     if (this.props.selectionType === "SINGLE_SELECT") {
-      isChanged = !(this.props.selectedOption.value === selectedOption.value);
+      isChanged =
```

Comment:
Since this is not modified at all, it can be declared as a const here
```suggestion
      const isChanged =
```

After removing the let declaration on line 102

Reply:
And a subtle issue with this logic as well. If the `this.props.selectedOption` is not present, the `isChanged` will be false and that will not trigger the value update down the line. We need to check for `this.props.selectedOption` before and that should not affect how `isChanged` is evaluated